spring/jta/src/main/java/org/jboss/narayana/quickstart/spring/config/DatabaseConfig.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.narayana.quickstart.spring.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+
+import javax.sql.DataSource;
+import java.util.Properties;
+
+@Configuration
+public class DatabaseConfig {
+    @Bean
+    public DataSource h2DataSource() {
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
+        dataSource.setDriverClassName("com.arjuna.ats.jdbc.TransactionalDriver");
+        dataSource.setUrl("jdbc:arjuna:h2:file:target/test.db");
```

Comment:
This setUrl is used in TransactionalDriver 
https://github.com/jbosstm/narayana/blob/master/ArjunaJTA/jdbc/classes/com/arjuna/ats/jdbc/TransactionalDriver.java#L88

Note - the url should start with "jdbc:arjuna"
we set the prop "DYNAMICAL_CLASS" 
```
prop.setProperty("DYNAMIC_CLASS", "org.jboss.narayana.quickstart.spring.config.H2DataSource");
```

https://github.com/jbosstm/narayana/blob/master/ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/DirectRecoverableConnection.java#L350
instance this class and call the getDataSource(String dbName)

So the setURL in the second place in the H2DataSource.getDataSource() method, it creates the real h2 xa datasource and it need to set the url with "jdbc"+dbName